pkg/kafka/auth/plaintext.go

```diff
@@ -15,17 +15,75 @@
 package auth
 
 import (
+	"crypto/sha256"
+	"crypto/sha512"
+	"fmt"
+	"hash"
+	"strings"
+
 	"github.com/Shopify/sarama"
+	"github.com/xdg/scram"
 )
 
+// XDGSCRAMClient is return a *sarama.SCRAMClient on create SCRAMClientGeneratorFunc when the mechanism is SCRAM-SHA-256 or SCRAM-SHA-512
```

Comment:
This sounds a bit strange. How about something like this?
```suggestion
// XDGSCRAMClient is the client to use when the auth mechanism is SCRAM
```

But if you don't have clashes anywhere, you can just use `SCRAMClient` instead.

Reply:
You might want to do a type assertion as well, like:

```go
var _ sarama.SCRAMClient = (*XDGSCRAMClient)(nil)
```

Reply:
I don't think this type even needs to be public, can just call it `type scramClient struct`...

Reply:
We still need the type assertion though :-)

Reply:
yes, I've already added it. :-)